units/include/scipp/units/unit.h

```diff
@@ -16,15 +16,20 @@ namespace scipp::units {
 class SCIPP_UNITS_EXPORT Unit {
 public:
   constexpr Unit() = default;
-  constexpr Unit(const llnl::units::precise_unit &u) noexcept : m_unit(u) {}
-  Unit(const std::string &unit);
+  constexpr explicit Unit(const llnl::units::precise_unit &u) noexcept
+      : m_unit(u) {}
+  explicit Unit(const std::string &unit);
 
-  constexpr auto underlying() const noexcept { return m_unit; }
+  [[nodiscard]] constexpr auto underlying() const noexcept { return m_unit; }
 
-  std::string name() const;
+  [[nodiscard]] std::string name() const;
 
-  bool isCounts() const;
-  bool isCountDensity() const;
+  [[nodiscard]] bool isCounts() const;
+  [[nodiscard]] bool isCountDensity() const;
+
+  [[nodiscard]] constexpr bool in_same_dimension(const Unit &other) const {
+    return m_unit.has_same_base(other.underlying());
+  }
```

Comment:
- No need for inlining?
- Is `base` or `dimension` the more correct terminology here? That is, why did you pick `same_dimension` rather than `same_base`?

Reply:
- It is inline because it is defined in the class definition. If you are talking about inlining the function _call_, then I doubt we need to persuade the compiler to do it with such a short function.
- Not sure. I typically think of it in terms of physical dimensions. But the word dimension is used for a lot of different things, in particular `units::Dim` is something different. Switching to `has_same_base`, if you agree?

Reply:
> It is inline because it is defined in the class definition.

Exactly, why do you need that?

Reply:
> Switching to has_same_base, if you agree?

👍

Reply:
> Exactly, why do you need that?

So you want to move it out-of-line? I put it inline because it is constexpr. Is there any change of doing some more complicated compile time computations with units in the future? If not, I can move the definition into the c++.

Reply:
I am asking because this header is essnetially included in *all* scipp compilation units, so unless we have a specific need for inlining we should not?

Reply:
Since `underlying` is inlined and `has_same_base` from `llnl::units` is provided inline, I don't think there will be any impact on compile performance. But I'm moving it out of line now because I don't have any real reason not to.